pkg/resolution/resolver/framework/interface.go

```diff
@@ -43,13 +43,17 @@ type Resolver interface {
 	// request and should return an error if any are missing or invalid.
 	ValidateParams(ctx context.Context, params []pipelinev1.Param) error
 
-	// Resolve receives the parameters passed via a resource request
+	// ValidateUrl is given the url from a resource
+	// request and should return an error if the resolver cannot resolve it.
+	ValidateUrl(ctx context.Context, url string) error
+
+	// Resolve receives the url and parameters passed via a resource request
 	// and returns the resolved data along with any annotations
 	// to include in the response. If resolution fails then an error
 	// should be returned instead. If a resolution.Error
 	// is returned then its Reason and Message are used as part of the
 	// response to the request.
-	Resolve(ctx context.Context, params []pipelinev1.Param) (ResolvedResource, error)
+	Resolve(ctx context.Context, url string, params []pipelinev1.Param) (ResolvedResource, error)
```

Comment:
What stability level are resolvers?
This is a backwards incompatible change for the interface. :(

You'll either need to define a new func or implement this as a variadic option.

Reply:
Same concern for func changes in `pkg/resolution/resource` and `pkg/remote/resolution`

Reply:
Good point. I will make those changes.

Reply:
Done!